File: 1456.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution{
public:
    int maxVowels(string s, int k){
        int curr = countVowels(s.substr(0, k));
        int max_vowels = curr;
        for(int i = k; i < s.size(); i++){
            curr += isVowel(s[i]) - isVowel(s[i - k]);
            max_vowels = max(max_vowels, curr);
        }
        return max_vowels;
    }
private:
    int countVowels(string s){
        int cnt = 0;
        for(int i = 0; i < s.size(); i++)
            cnt += isVowel(s[i]);
        return cnt;
    }
    bool isVowel(char c){
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
};

int main(){
    string s;
    int k;
    cin >> s >> k;
    Solution s;
    cout << s.maxVowels(s, k) << endl;
    return 0;
}

File: python/1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        curr = sum(1 for c in s[:k] if c in 'aeiou')
        max_vowels = curr
        for i in range(k, len(s)):
            curr += s[i] in 'aeiou'
            curr -= s[i - k] in 'aeiou'
            max_vowels = max(max_vowels, curr)
        return max_vowels
    
if __name__ == '__main__':
    s = input()
    k = int(input())
    print(Solution().maxVowels(s, k))